cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -471,6 +478,46 @@ public void onBecomeFollowerFromOffline(Message message, NotificationContext con
           Utils.restoreLocalDBFromS3(adminPort, dbName, s3Bucket, s3Path, snapshotHost,
               snapshotPort);
         }
+
+        try {
+          LOG.error(dbName + " Catching up live updates to leader");
+          Map<String, String> liveHostAndRole = getLiveHostAndRole(context, dbName);
+          // Find upstream, prefer Leader
+          String upstream = null;
+          for (Map.Entry<String, String> instanceNameAndRole : liveHostAndRole.entrySet()) {
+            String role = instanceNameAndRole.getValue();
+            if (role.equalsIgnoreCase("LEADER")) {
+              upstream = instanceNameAndRole.getKey();
+              break;
+            } else {
+              upstream = instanceNameAndRole.getKey();
```

Comment:
if somehow, you returned a `liveHostAndRole` in order: follower, follower, offline. then, won't this logic set the upstream to offline?

Reply:
`getLiveHostAndRole` does not return offline replicas.

Reply:
it does? 
```
      if (!role.equalsIgnoreCase("LEADER") &&
          !role.equalsIgnoreCase("FOLLOWER") &&
          !role.equalsIgnoreCase("OFFLINE")) {
        continue;
      }
```

Reply:
Hmm, I read that wrong. 
This is the same code which we use to choose a leader to backup and restore from. 
Let me read it again and see why it's being done this way.

Reply:
I've added a check to skip any offline replicas.